modules/search/client/services/search.services.js

```diff
@@ -0,0 +1,129 @@
+'use strict';
+
+angular.module('search')
+	.service('SearchService', searchService)
+	.service('SearchResultsService', searchResultsService);
+
+searchService.$inject = [ '$http', '$state', '$rootScope', '$timeout', '_', 'ProjectModel', 'TreeModel'];
+/* @ngInject */
+function searchService( $http, $state, $rootScope, $timeout, _, ProjectModel, TreeModel) {
+	var self = this;
+	self.searchResults = {};
+	self.searchText = '';
+	self.rootNode = null;
+
+	return {
+		getSearchText: getSearchText,
+		getSearchResults: getSearchResults,
+		setupDirectoryTree: setupDirectoryTree,
+		redirectSearchDocuments: redirectSearchDocuments,
+		composeFilePath: composeFilePath,
+		searchDocuments: searchDocuments
+	};
+
+	function setSearchText(searchText) {
+		self.searchText = searchText;
+		$timeout(function() {
+			$rootScope.$broadcast('search-text-changed');
+		},100);
+	}
+
+	function getSearchText () {
+		return self.searchText;
+	}
+
+	function getSearchResults () {
+		return self.searchResults;
+	}
+
+	function redirectSearchDocuments(project, searchText, start, limit, orderBy, direction, collection) {
+		//console.log("transition to search with searchText", searchText, start, limit, orderBy);
+		setSearchText(searchText);
+		$state.go('p.search', {
+			projectid: project.code,
+			searchText: searchText,
+			collection: collection,
+			start: start,
+			limit: limit,
+			orderBy: orderBy,
+			direction: direction
+		});
+	}
+
+	function setupDirectoryTree (project) {
+		var tree = new TreeModel();
+		return ProjectModel.getProjectDirectory(project)
+		.then(function (dir) {
+			project.directoryStructure = dir || {
+					id: 1,
+					lastId: 1,
+					name: 'ROOT',
+					published: true
+				};
+			self.rootNode = tree.parse(project.directoryStructure);
+			return self.rootNode;
+		});
+	}
+
+	function composeFilePath(directoryID, fileName) {
+		var path = '';
+		if (self.rootNode) {
+			var theNode = self.rootNode.first(function (n) {
+				return n.model.id === directoryID;
+			});
+			var pathSet = theNode ? theNode.getPath() || [] : [];
+			_.forEach(pathSet, function (element) {
+				var n = element.model.name;
+				if (n !== 'ROOT') {
```

Comment:
This code is copy/paste from docManager. Is there a better sample I should use?  I can only find isRoot on the project object not document object